Fix mypy (fixes CI style check)

azat opened the pull request:
I've inspected the last image that works (from
https://github.com/ClickHouse/ClickHouse/pull/53493) and the one that does not, and downgrading PyGithub helps (though at first I thought that the culprit was typing_extensions)

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix mypy (fixes CI style check)

Follow-up for: https://github.com/ClickHouse/ClickHouse/pull/51769 (cc @Felixoid )

azat committed:
Fix mypy

I've inspected the last image that works (from
https://github.com/ClickHouse/ClickHouse/pull/53493) and the one that
does not, and downgrading PyGithub helps (though at first I thought that
the culprit was typing_extensions)

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

robot-clickhouse-ci-1 commented:
<!-- automatic status comment for PR #55227 from azat/ClickHouse:mypy-fix -->
*This is an automated comment for commit f2e4fc2206b942a34a5ec938c2ebc50ac1aafe96 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55227/f2e4fc2206b942a34a5ec938c2ebc50ac1aafe96/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6404409526">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55227/f2e4fc2206b942a34a5ec938c2ebc50ac1aafe96/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

azat commented:
>Style Check — Check mypy failed.

@Felixoid it does not uses the per-PR images:

```
INFO:root:Is going to run the command: docker run -u $(id -u ${USER}):$(id -g ${USER}) --cap-add=SYS_PTRACE --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse:/ClickHouse --volume=/home/ubuntu/actions-runner/_work/_temp/style_check/style_check:/test_output clickhouse/style-test:latest
...
```

While if you run manually:

```
$ docker run --rm -v $PWD:/src:ro -it clickhouse/style-test:55227 bash
root@c2145fbfb708:/src# mypy --cache-dir /tmp/mypy-cache --config-file=tests/ci/.mypy.ini tests/ci/github_helper.py
Success: no issues found in 1 source file
```

azat committed:
Fix using of per-PR docker images

    Artifact changed_images was downloaded to /home/ubuntu/actions-runner/_work/_temp/style_check
    ...
    INFO:root:User robot-ch-test-poll4 with 4980 remaining requests is used
    INFO:root:Looking for images file in /home/ubuntu/actions-runner/_work/_temp/style_check/style_check
    INFO:root:Images file not found
    INFO:root:Pulling image clickhouse/style-test:latest
    INFO:root:Is going to run the command: docker run -u $(id -u ${USER}):$(id -g ${USER}) --cap-add=SYS_PTRACE --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse:/ClickHouse --volume=/home/ubuntu/actions-runner/_work/_temp/style_check/style_check:/test_output clickhouse/style-test:latest

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
>PullRequestCI / StyleCheck (pull_request) Successful in 6m

Now passed, other checks should not be touched by this patches.

azat commented:
Closing in favor of #55228

Felixoid commented:
THanks!

Sorry, didn't see your PR, otherwise I'd have it merged instead of even starting my own
